frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -204,35 +204,64 @@ fn decl_all_pallets<'a>(
 		types.extend(type_decl);
 		names.push(&pallet_declaration.name);
 	}
-	// Make nested tuple structure like (((Babe, Consensus), Grandpa), ...)
+
+	// Make nested tuple structure like:
+	// `((FirstPallet, (SecondPallet, ( ... , LastPallet) ... ))))`
 	// But ignore the system pallet.
-	let all_pallets = names
+	let all_pallets_without_system = names
 		.iter()
 		.filter(|n| **n != SYSTEM_PALLET_NAME)
+		.rev()
 		.fold(TokenStream2::default(), |combined, name| quote!((#name, #combined)));
 
+	// Make nested tuple structure like:
+	// `((FirstPallet, (SecondPallet, ( ... , LastPallet) ... ))))`
 	let all_pallets_with_system = names
+		.iter()
+		.rev()
+		.fold(TokenStream2::default(), |combined, name| quote!((#name, #combined)));
+
+	// Make nested tuple structure like:
+	// `((LastPallet, (SecondLastPallet, ( ... , FirstPallet) ... ))))`
+	// But ignore the system pallet.
+	let all_pallets_without_system_reversed = names
+		.iter()
+		.filter(|n| **n != SYSTEM_PALLET_NAME)
+		.fold(TokenStream2::default(), |combined, name| quote!((#name, #combined)));
+
+	// Make nested tuple structure like:
+	// `((LastPallet, (SecondLastPallet, ( ... , FirstPallet) ... ))))`
+	let all_pallets_with_system_reversed = names
 		.iter()
 		.fold(TokenStream2::default(), |combined, name| quote!((#name, #combined)));
 
 	quote!(
 		#types
 
 		/// All pallets included in the runtime as a nested tuple of types.
-		/// Excludes the System pallet.
-		pub type AllPallets = ( #all_pallets );
+		#[deprecated(note = "The type definition has changed from representing all pallets \
+			excluding system, in reversed order to become the representation of all pallets \
+			including system pallet in regular order. For this reason it is encouraged to use \
+			explicitly one of `AllPalletsWithSystem`, `AllPalletsWithoutSystem`, \
+			`AllPalletsWithSystemReversed`, `AllPalletsWithoutSystemReversed`. \
+			Note that the type `frame_executive::Executive` expects one of `AllPalletsWithSystem` \
+			and `AllPalletsWithSystemReversed. More details in \
+			https://github.com/paritytech/substrate/pull/10043")]
+		pub type AllPallets = AllPalletsWithSystem;
```

Comment:
This is wrong right?

If you are going to provide the old `AllPallets` type, it should have the same behavior has before?